application/src/main/java/org/togetherjava/tjbot/commands/utils/MessageUtils.java

```diff
@@ -56,4 +59,25 @@ public static String escapeMarkdown(String text) {
         return afterEscape.replace("\\```", "\\`\\`\\`");
     }
 
+    /**
+     * Converts a slash command text to a mentioned slash command, which you can directly click on
+     * in Discord.
+     *
+     * @param guild the {@link Guild} that contains the command
+     * @param command the command's name
+     * @return Formatted string for the mentioned slash command
+     * @throws IllegalArgumentException when the command name doesn't match with the guild's
+     *         commands
+     */
+    public static RestAction<String> mentionSlashCommand(Guild guild, String command) {
+        return guild.retrieveCommands().map(commands -> {
+            Command foundCommand = commands.stream()
+                .filter(c -> c.getName().equalsIgnoreCase(command))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(
+                        "Command %s does not exist in guild %s".formatted(command, guild.getId())));
+            return String.format("</%s:%d>", command, foundCommand.getIdLong());
```

Comment:
never rely on the toString representation of objects. JDA might change in the next update what `command.toString()` gives and then this will break.

Instead, build the string you expect urself. I suspect u intended to do `command.getName()` here or similar.

Reply:
I'd personally also use toString here, at least, what is the purpose for toString if it's not logging?

Reply:
im not talking about the log/error message. but the `String.format(..., command, ...)`, which is application logic.

Reply:
OWH nvm yes

the to string method returns the type & name & id, so does that even work?

Reply:
The `command` parameter here comes from the static field `COMMAND_NAME` in the caller's classes, so it's not related to JDA. I think the relevant thing to do would be the change the field's name to something more precise

Reply:
ah `command` vs `foundCommand`... yeah... prime example of bad variable naming haha. u got us both 😝

Reply:
Naming fields is always difficult 😅
I'll try to find something more relevant when I have time to work on that again